src/annotator/integrations/html-metadata.js

```diff
@@ -183,27 +151,31 @@ export class HTMLMetadata {
     return tags;
   }
 
-  _getTitle() {
-    if (this.metadata.highwire.title) {
-      this.metadata.title = this.metadata.highwire.title[0];
-    } else if (this.metadata.eprints.title) {
-      this.metadata.title = this.metadata.eprints.title[0];
-    } else if (this.metadata.prism.title) {
-      this.metadata.title = this.metadata.prism.title[0];
-    } else if (this.metadata.facebook.title) {
-      this.metadata.title = this.metadata.facebook.title[0];
-    } else if (this.metadata.twitter.title) {
-      this.metadata.title = this.metadata.twitter.title[0];
-    } else if (this.metadata.dc.title) {
-      this.metadata.title = this.metadata.dc.title[0];
+  /** @param {HTMLDocumentMetadata} metadata */
+  _getTitle(metadata) {
+    if (metadata.highwire.title) {
+      return metadata.highwire.title[0];
+    } else if (metadata.eprints.title) {
+      return metadata.eprints.title[0];
+    } else if (metadata.prism.title) {
+      return metadata.prism.title[0];
+    } else if (metadata.facebook.title) {
+      return metadata.facebook.title[0];
+    } else if (metadata.twitter.title) {
+      return metadata.twitter.title[0];
+    } else if (metadata.dc.title) {
+      return metadata.dc.title[0];
     } else {
-      this.metadata.title = this.document.title;
+      return this.document.title;
     }
   }
 
-  _getLinks() {
+  /**
+   * @param {Pick<HTMLDocumentMetadata, 'highwire'|'dc'>} metadata
+   */
+  _getLinks(metadata) {
     // We know our current location is a link for the document.
-    this.metadata.link = [{ href: this._getDocumentHref() }];
+    const links = [{ href: this._getDocumentHref() }];
 
     // Extract links from certain `<link>` tags.
```

Comment:
Again, inherited, but what does it mean by _certain_?

Reply:
I've clarified this comment to make it clear that this means _links with certain "rel" values_.